internal/healthcheck/healthcheck_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthcheck
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/proxy"
+)
+
+const testPort = "8080"
+
+// Test to verify that when the proxy client is up, the liveness endpoint writes 200.
+func TestLiveness(t *testing.T) {
+	proxyClient := &proxy.Client{}
+	hc := NewHealthCheck(proxyClient, testPort)
+	defer hc.Close(context.Background()) // Close health check upon exiting the test.
+
+	resp, err := http.Get("http://localhost:" + hc.port + livenessPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if resp.StatusCode != 200 {
+		t.Errorf("got status code %v instead of 200", resp.StatusCode)
+	}
+}
+
+// Test to verify that when startup has not finished, the readiness endpoint writes 500.
+func TestStartupFail(t *testing.T) {
```

Comment:
nit: You may wish to consider table driven tests, which is fairly idiomatic to Go: https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go

Reply:
Cool! Is it possible to use table-driven tests when the underlying conditions in each test differ? I.e., some of the tests call `NotifyStarted()`, the proxy client is set up differently in `TestMaxConnectionsReached`.

Reply:
I wouldn't combine all your tests into a single table driven test, but would instead combine similar tests. One example in the proxy is [this](https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/97981d1e90b5df3f52be41d0944f10f5442932e0/cmd/cloud_sql_proxy/proxy_test.go#L37).

So I think `TestStartupFail` and `TestStartupPass` would be good candidates here, as they test the same feature and have a large overlap in code.